editor/src/components/canvas/controls/position-outline.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react'
+import { getLayoutProperty } from '../../../core/layout/getLayoutProperty'
+import { MetadataUtils, PropsOrJSXAttributes } from '../../../core/model/element-metadata-utils'
+import { eitherToMaybe, isRight, left, right } from '../../../core/shared/either'
+import * as EP from '../../../core/shared/element-path'
+import { isJSXElement } from '../../../core/shared/element-template'
+import { CanvasPoint, CanvasRectangle } from '../../../core/shared/math-utils'
+import { ElementPath } from '../../../core/shared/project-file-types'
+import { useColorTheme } from '../../../uuiui'
+import { betterReactMemo } from '../../../uuiui-deps'
+import { useEditorState } from '../../editor/store/store-hook'
+
+interface PositionOutlineProps {
+  path: ElementPath
+  frame: CanvasRectangle
+}
+
+export const PositionOutline = betterReactMemo('PositionOutline', (props: PositionOutlineProps) => {
+  const containingFrame = useContainingFrameForElement(props.path)
+  const attributes = usePropsOrJSXAttributes(props.path)
+  const canvasOffset = useEditorState(
+    (store) => store.editor.canvas.roundedCanvasOffset,
+    'PositionOutline canvasOffset',
+  )
+  if (containingFrame != null) {
+    let pins: PinOutlineProps[] = useCollectPinOutlines(
```

Comment:
We should never use hooks inside an `if` statement, _but_ `useCollectPinOutlines` isn't actually a hook so it can just be renamed.

Reply:
oops, fixed :)